### PR TITLE
Add section about doc testing

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Adds section about how to write doc examples and how to add them to the doctests.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -283,3 +283,71 @@ We have an automatic script running with the `make style` comment that will make
...
+    >>> # audio file is decoded on the fly
+    >>> inputs = processor(dataset[0]["audio"]["array"], sampling_rate=sampling_rate, return_tensors="pt")
+    >>> with torch.no_grad():
+    ...     logits = model(**inputs).logits
+    >>> predicted_ids = torch.argmax(logits, dim=-1)
+
+    >>> # transcribe speech
+    >>> transcription = processor.batch_decode(predicted_ids)
+    >>> transcription[0]
+	'MISTER QUILTER IS THE APOSTLE OF THE MIDDLE CLASSES AND WE ARE GLAD TO WELCOME HIS GOSPEL'
```
Comment (sgugger) [inline docs/README.md:323]:
```suggestion
    'MISTER QUILTER IS THE APOSTLE OF THE MIDDLE CLASSES AND WE ARE GLAD TO WELCOME HIS GOSPEL'
```

File: docs/README.md [hunk 2]
```diff
@@ -283,3 +283,71 @@ We have an automatic script running with the `make style` comment that will make
...
+[docstring of its forward function](https://huggingface.co/docs/transformers/model_doc/wav2vec2#transformers.Wav2Vec2ForCTC.forward).
+
+## Writing documenation examples
+
+The syntax for Example docstrings can look as follows:
+
+```
+    Example:
+
+    \`\`\`python
```
Comment (sgugger) [inline docs/README.md:302]: I think this will be rendered with the \\, so you should remove them.

File: docs/README.md [hunk 3]
```diff
@@ -283,3 +283,71 @@ We have an automatic script running with the `make style` comment that will make
...
+    >>> inputs = processor(dataset[0]["audio"]["array"], sampling_rate=sampling_rate, return_tensors="pt")
+    >>> with torch.no_grad():
+    ...     logits = model(**inputs).logits
+    >>> predicted_ids = torch.argmax(logits, dim=-1)
+
+    >>> # transcribe speech
+    >>> transcription = processor.batch_decode(predicted_ids)
+    >>> transcription[0]
+	'MISTER QUILTER IS THE APOSTLE OF THE MIDDLE CLASSES AND WE ARE GLAD TO WELCOME HIS GOSPEL'
+    \`\`\`
```
Comment (sgugger) [inline docs/README.md:324]: Same as above.

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

File: docs/README.md (hunk 2 above)
Comment (patrickvonplaten) [inline docs/README.md:302]:
```suggestion
    ```python
```

File: docs/README.md (hunk 3 above)
Comment (patrickvonplaten) [inline docs/README.md:324]:
```suggestion
    ```
```
